packages/runtime/datastore/src/channelStorageService.ts

```diff
@@ -50,7 +50,16 @@ export class ChannelStorageService implements IChannelStorageService {
             ? (await this.extraBlobs).get(id)
             : undefined;
 
-        return blob ?? this.storage.read(id);
+        return blob ?? this.storage.readString(id);
```

Comment:
read() should never call directly readString (and vice versa) without extra modifications, as they work with different formats (read - with base64 encoded data, likely binary, and readString - with strings)